worlds/yugioh06/__init__.py

```diff
@@ -0,0 +1,538 @@
+            entrance.connect(region)
+            self.multiworld.regions.append(region)
+
+        card_shop = self.multiworld.get_region('Card Shop', self.player)
+        # Booster Contents
+        for booster in booster_packs:
+            region = create_region(self,
+                                   booster, get_booster_locations(booster))
+            entrance = Entrance(self.player, booster, card_shop)
+            entrance.access_rule = (lambda unlock: lambda state: state.has(unlock, self.player))(booster)
+            card_shop.exits.append(entrance)
+            entrance.connect(region)
+            self.multiworld.regions.append(region)
+
+        challenge_region = self.multiworld.get_region('Challenges', self.player)
+        # Challenges
+        for challenge, lid in (Limited_Duels | Theme_Duels).items():
+            if challenge in self.removed_challenges:
+                continue
+            region = create_region(self,
+                                   challenge, {challenge: lid, challenge + " Complete": None})
+            entrance = Entrance(self.player, challenge, challenge_region)
+            entrance.access_rule = (lambda unlock: lambda state: state.has(unlock + " Unlock", self.player))(challenge)
+            challenge_region.exits.append(entrance)
+            entrance.connect(region)
+            self.multiworld.regions.append(region)
+
+    def generate_early(self):
+        if self.options.structure_deck.current_key == "none":
+            boosters = draft_boosters
+            if self.options.campaign_opponents_shuffle.value:
+                opponents = tier_1_opponents
+            else:
+                opponents = draft_opponents
+        else:
+            boosters = booster_packs
+            opponents = tier_1_opponents
+
+        if self.options.structure_deck.current_key == "random_deck":
+            self.options.structure_deck.value = self.random.choice([0, 1, 2, 3, 4, 5])
+        for item in self.options.start_inventory:
+            if item in opponents:
+                self.starting_opponent = item
+            if item in boosters:
+                self.starting_booster = item
+        if not self.starting_opponent:
+            self.starting_opponent = self.multiworld.random.choice(opponents)
+        self.multiworld.push_precollected(self.create_item(self.starting_opponent))
+        if not self.starting_booster:
+            self.starting_booster = self.multiworld.random.choice(boosters)
+        self.multiworld.push_precollected(self.create_item(self.starting_booster))
+        banlist = self.options.banlist.value
+        self.multiworld.push_precollected(self.create_item(Banlist_Items.get(banlist)))
+        challenge = list((Limited_Duels | Theme_Duels).keys())
+        noc = len(challenge) - max(self.options.third_tier_5_campaign_boss_challenges.value
+                                   if self.options.third_tier_5_campaign_boss_unlock_condition.value == 1 else 0,
```

Comment:
can switch this (and the rest in the block) to 
`if self.options.third_tier_5_campaign_boss_unlock_condition == "challenges" else 0,`

Reply:
applied suggestion